spharpy/spherical.py

```diff
@@ -767,3 +768,404 @@ def sph_identity_matrix(n_max, type='n-nm'):
+    It can be either 'complex' or 'real'.
+        - ``'complex'``: Uses complex-valued spherical harmonics.
+        - ``'real'``: Uses real-valued spherical harmonics.
+
+    - normalization: Defines the normalization convention.
+     It can be 'n3d', 'maxN', or 'sn3d'.
+        - ``'n3d'``: Uses the 3D normalization
+        (also known as Schmidt semi-normalized).
+        - ``'maxN'``: Uses the maximum norm
+        (also known as fully normalized).
+        - ``'sn3d'``: Uses the SN3D normalization
+        (also known as Schmidt normalized).
+
+    - channel_convention: Defines the channel ordering convention.
+    It can be either 'acn' or 'fuma'.
+        - ``'acn'``: Follows the Ambisonic Channel Number (ACN) convention.
+        - ``'fuma'``: Follows the Furse-Malham (FuMa) convention.
+
+    - inverse_transform: Defines the type of inverse transform.
+     It can be 'pseudo_inverse', 'quadrature', or None.
+        - ``'pseudo_inverse'``: Uses the Moore-Penrose pseudo-inverse
+         for the inverse transform.
+        - ``'quadrature'``: Uses quadrature for the inverse transform.
+        - ``None``: No inverse transform is applied.
+
+    Parameters
+    ----------
+    n_max : int
+        Maximum spherical harmonic order
+    coordinates : :doc:`pf.Coordinates <pyfar:classes/pyfar.coordinates>` or
+        `sp.SamplingSphere <spharpy:classes/spharpy.samplings.coordinates>`
+        objects with sampling points for which the basis matrix is
+        calculated
+    basis_type : str, optional
+        Type of spherical harmonic basis, either ``'complex'`` or
+        ``'real'``. The default is ``'complex'``.
+    normalization : str, optional
+        Normalization convention, either ``'n3d'``, ``'maxN'`` or
+         ``'sn3d'``. The default is ``'n3d'``.
+        (maxN is only supported up to 3rd order)
+    channel_convention : str, optional
+        Channel ordering convention, either ``'acn'`` or ``'fuma'``.
+        The default is ``'acn'``.
+        (FuMa is only supported up to 3rd order)
+    inverse_transform : str, optional
+        Inverse transform type, either ``'pseudo_inverse'`` or
+        ``'quadrature'``. The default is None.
+    phase_convention : string or None, optional
+        Whether to include the Condon-Shortley phase term.
+        The default is None.
+    """
+
+    def __init__(
+        self,
+        n_max,
+        coordinates,
+        basis_type="real",
+        normalization="n3d",
+        channel_convention="acn",
+        inverse_transform=None,
+        phase_convention=None
+    ):
+        # hidden attributes
+        self._n_max = None
+        self._coordinates = pf.Coordinates()
+        self._basis_type = None
+        self._weights = None
+        self._inverse_transform = None
+        self._channel_convention = None  # ch. ord. conv.
+        self._normalization = None  # gain norm. conv.
+        self._phase_convention = None
+
+        self.n_max = n_max
```

Comment:
we need a check that when a user inits a signal with n_max > 3, fuma and maxN is not allowed. I think that is not caught anywhere yet. Probably the best place are the setters?